src/exceptions/clientException.ts

```diff
@@ -1,29 +1,24 @@
+import { AxiosError } from 'axios';
 import { Exception } from './exception';
 
 export class ClientException extends Exception {
   private static readonly statusCodeMap: Record<number, number> = {
     400: 422,
-    401: 401,
-    403: 403,
     404: 422,
   };
 
-  constructor(serviceName: string, error?: unknown) {
-    super('Dependent service returned error', ClientException.convertStatusCode(error as object), {
+  constructor(serviceName: string, error: AxiosError) {
```

Comment:
At this point, should we rename the `ClientException` to `AxiosException` so it is obvious where it comes from?

Reply:
I would still keep the name generic.. in case of things that won't happen haha. Maybe ClientException <AxiosError> ? IDK if that would really work tho

Also, let's have the second parameter optional. In case you wanna construct this without an axios error

Reply:
Changing it to non-optional and renaming it to `AxiosException` is something I wanted to bind together... So that it cannot be created without an axios error.

Is there any benefit in having this as non-axios related exception?

Reply:
If you e.g. don't use our client / axios to make that call. I could imagine using other ppl client libraries, but also what are the chances

Reply:
True, but this library already needs to at least peer depend on Axios to pull the `AxiosError` type in.

If we want to make it truly independent... as this is typescript .. what about defining a compatible type here? I.e. one that has some of the `AxiosError` properties (starting with `response.status`).

Something like `HttpError`?

Reply:
Screw it let's leave it as `AxiosError ` then and see how far we will get :)

Reply:
Where is the over-engineering spirit? 🥳

Reply:
Since none of it is a breaking change.. I'm thinking about releasing it as `2.0.1` or `2.1.0` that also works for non-breaking changes.

`2.1.0` requires a new version of AT-HttpClient and AT-Exceptions because we locked those to `2.0.x` .. it wasn't a wise choice! :D should really be blocked only against breaking major version changes.